### PR TITLE
[ENG-4421] No BsForm

### PR DESCRIPTION
**Pull request by brianjgeiger**
-   Ticket: [ENG-4421]
-   Feature flag: n/a

## Purpose

Remove all the `BsForm`s, which also involves removing all the `ValidatedModelForm`s, which are based on `BsForm`. Either just get rid of them or replace them with `FormControls`

## Summary of Changes

1. Remove unused settings.profile.name form
2. Remove validated model form from handbook docs
3. Remove under-used form from comment card
4. Update form for zoom-to-route
5. Update the forms for settings.tokens
6. Remove the `ValidatedModelForm`

## Side Effects

I upgraded the zoom-to-route functionality to allow you to search the routes rather than just scrolling through the whole list each time. I also threw in some error handling for the tokens stuff.

## QA Notes

The main thing to check is if you can create and edit Personal Access Tokens still.

[ENG-4421]: https://openscience.atlassian.net/browse/ENG-4421?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by adlius — APPROVED** [review]
LGTM. Just some questions.

File: app/settings/profile/name/-components/name/template.hbs [hunk 1]
```diff
@@ -1,62 +0,0 @@
-<ValidatedModelForm
```
[the remark this answers is not in the document]
Comment (brianjgeiger) [reply app/settings/profile/name/-components/name/template.hbs:1]: Yeah, this was part of the embosf project to convert over the whole site. I guess I was working on the settings/profile page when I was pulled off the project, and we never got back to it. One day we may come back to work on this, but we'll want to use a more modern form for it in any case.

File: tests/acceptance/settings/tokens-page-test.ts [hunk 2]
```diff
@@ -59,7 +60,7 @@ module('Acceptance | settings | personal access tokens', hooks => {
         await untrackedClick('[data-test-scope] input[type=checkbox]');
         await percySnapshot(assert);
         await click('[data-analytics-name="Submit button"]');
-
+        await timeout(50);
```
[the remark this answers is not in the document]
Comment (brianjgeiger) [reply tests/acceptance/settings/tokens-page-test.ts:63]: This is a consequence of the [removal of the jQuery integration](https://github.com/CenterForOpenScience/ember-osf-web/pull/1831). Doing async things sometimes takes time in rendering, and it's currently unclear under what circumstances it's going to cause problems for tests. `await settled()` does nothing to solve the problem. If it's just loading data at the beginning, you can pre-load the data, or you can add a small delay before the asserts, which is the only thing I could figure out what to do in the case of mutation or addition of data.

[the remark this answers is not in the document]
Comment (adlius) [reply tests/acceptance/settings/tokens-page-test.ts:63]: I checked ember-test-helpers and it seems that `settled()` does not listen for Fetch API requests. I'll dig a bit deeper and see whether there is anything we can do to improve ember-test-helpers.

[the remark this answers is not in the document]
Comment (brianjgeiger) [reply tests/acceptance/settings/tokens-page-test.ts:63]: Awesome, thank you.

**Pull request merged**